packages/web-components/src/pane/pane.ts

```diff
@@ -0,0 +1,260 @@
+import { attr, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { isHTMLElement, keyEscape, keyTab } from '@microsoft/fast-web-utilities';
+import { PanePosition, PaneSize } from './pane.options.js';
+
+export interface OpenEvent {
+  open: boolean;
+  position?: string;
+  controlSize?: PaneSize | number;
+}
+
+// The Pane class extends the FASTElement which provides a foundation for creating
+// custom elements using FAST. It extends the HTMLElement interface of the browser.
+export class Pane extends FASTElement {
+  private panes: Pane[] = [];
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    // Register event listener
+    this.addEventListener('keydown', this.handleKeyDown);
+    this.setItems();
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    // Unregister event listener
+    this.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  private hasInteracted: boolean = false;
+
+  // The previous active element before opening the pane.
+  private previousActiveElement?: HTMLElement;
+
+  /**
+   * The pane element.
+   * @public
+   * @remarks
+   * HTML Attribute: pane
+   */
+  @observable
+  public _pane?: HTMLElement;
+
+  /**
+   * Determines whether the pane should be displayed as modal or non-modal.
+   * When in modal mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modal
+   */
+  @attr({ mode: 'boolean' })
+  public modal: boolean = false;
+
+  /**
+   * Sets the position of the pane (left/right).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue right
+   */
+  @attr
+  public position?: PanePosition;
+
+  /**
+   * The maximum number of options to display.
+   *
+   * @remarks
+   * HTML Attribute: `size`.
+   *
+   * @public
+   */
+  @attr({ attribute: 'control-size' })
+  public controlSize?: PaneSize | number | undefined;
```

Comment:
string, arbitrary number or undefined? Should we really allow arbitrary numbers and set via JS or should we prefer CSS?